graph/src/blockchain/block_stream.rs

```diff
@@ -441,32 +451,12 @@ async fn get_entities_for_range(
     from: BlockNumber,
     to: BlockNumber,
 ) -> Result<BTreeMap<BlockNumber, Vec<EntityWithType>>, Error> {
-    let mut entities_by_block = BTreeMap::new();
-
+    let mut entity_types = vec![];
     for entity_name in &filter.entities {
         let entity_type = schema.entity_type(entity_name)?;
-
-        let entity_ranges = store.get_range(&entity_type, from..to)?;
-
-        for (block_number, entity_vec) in entity_ranges {
-            let mut entity_vec = entity_vec
-                .into_iter()
-                .map(|e| EntityWithType {
-                    entity_type: entity_type.clone(),
-                    entity: e,
-                })
-                .collect();
-
-            entities_by_block
-                .entry(block_number)
-                .and_modify(|existing_vec: &mut Vec<EntityWithType>| {
-                    existing_vec.append(&mut entity_vec);
-                })
-                .or_insert(entity_vec);
-        }
+        entity_types.push(entity_type);
     }
```

Comment:
It would be more idiomatic to write this as `let entity_types = filter.entities.iter().map(|name| schema.entity_type(name)).collect()?`

Reply:
That worked baring the trailing error propagation operator.